flink-examples/flink-examples-streaming/src/main/java/org/apache/flink/streaming/examples/gpu/MatrixVectorMul.java

```diff
@@ -0,0 +1,258 @@
+		} else {
+			System.out.println(String.format("Executing MatrixVectorMul example with default data size %d.\nUse --data-size to specify size of the vector stream.", DEFAULT_DATA_SIZE));
+			dataSize = DEFAULT_DATA_SIZE;
+		}
+
+		String resourceName;
+		if (params.has("resource-name")) {
+			resourceName = params.get("resource-name");
+		} else {
+			System.out.println(String.format("Executing MatrixVectorMul example with default resource name %s.\nUse --resource-name to specify resource name of GPU.", DEFAULT_RESOURCE_NAME));
+			resourceName = DEFAULT_RESOURCE_NAME;
+		}
+
+		DataStream<List<Float>> result = env.addSource(new RandomVectorSource(dimension, dataSize))
+						.map(new Multiplier(dimension, resourceName));
+
+		// Emit result
+		if (params.has("output")) {
+			result.addSink(StreamingFileSink.forRowFormat(new Path(params.get("output")),
+					new SimpleStringEncoder<List<Float>>()).build());
+		} else {
+			System.out.println("Printing result to stdout. Use --output to specify output path.");
+			result.print();
+		}
+		// Execute program
+		env.execute("Matrix-Vector Multiplication");
+	}
+
+	// *************************************************************************
+	// USER FUNCTIONS
+	// *************************************************************************
+
+	/**
+	 * Random vector source which generates random vectors with specified dimension and total data size.
+	 */
+	private static final class RandomVectorSource extends RichSourceFunction<List<Float>> {
+
+		private transient volatile boolean running;
+		private final int dimension;
+		private final int dataSize;
+
+		RandomVectorSource(int dimension, int dataSize) {
+			this.dimension = dimension;
+			this.dataSize = dataSize;
+		}
+
+		@Override
+		public void open(Configuration parameters) {
+			running = true;
+		}
+
+		@Override
+		public void run(SourceContext<List<Float>> ctx) {
+			int count = 0;
+			while (running && count < dataSize) {
+				List<Float> randomRecord = new ArrayList<>();
+				for (int i = 0; i < dimension; ++i) {
+					randomRecord.add((float) Math.random());
+				}
+				ctx.collect(randomRecord);
+				count += 1;
+			}
+		}
+
+		@Override
+		public void cancel() {
+			running = false;
+		}
+	}
+
+	/**
+	 * Matrix-Vector multiplier using CUBLAS library.
+	 */
+	private static final class Multiplier extends RichMapFunction<List<Float>, List<Float>> {
+		private final int dimension;
+		private final String resourceName;
+		private Pointer matrixPointer;
+
+		Multiplier(int dimension, String resourceName) {
+			this.dimension = dimension;
+			this.resourceName = resourceName;
+		}
+
+		@Override
+		public void open(Configuration parameters) {
+			// When multiple instances of this class and JCuda exist in different class loaders, then we will get UnsatisfiedLinkError.
+			// To avoid that, we need to temporarily override the java.io.tmpdir, where the JCuda store its native library, with a random path.
```

Comment:
Can you explain the problem a bit more in detail? I would like to understand what's the underlying problem. Is there an official issue describing the problem?

Reply:
Yes, please refer to https://issues.apache.org/jira/browse/FLINK-5408

Reply:
Another reference: http://mail-archives.apache.org/mod_mbox/flink-user/201805.mbox/%3CCAMUrgW-GqNVyvMiL5Th56Hxm6eoDiM=u_DB59EBN-15ZqgMTyw@mail.gmail.com%3E

Reply:
Could you add these pointers to the in-code comments so that other devs can find it?

Reply:
Could you try whether this problem still exists because we have changed how user code class loaders are instantiated and how long the live with the 1.11 release (see https://issues.apache.org/jira/browse/FLINK-16408)

Reply:
I check the git history of `gpu-case` branch and I afraid that the issue will not be fixed by it. IIRC, different jobs (in session mode) will use different class loaders (and thus load the native library multiple times) anyways.